File: 1402-reducing-dishes/1402-reducing-dishes.cpp
class Solution {
public:
    int maxSatisfaction(vector<int>& s) {
        int ans =0,res =0;
        sort(begin(s),end(s));
        for(int i=s.size()-1;i>=0 and res+s[i]>0;--i){
            res+=s[i];
            ans +=(res);
        }
        return ans;
    }
};

File: 1402-reducing-dishes/1402-reducing-dishes.py
class Solution:
    def maxSatisfaction(self, s: List[int]) -> int:
        s.sort()
        ans =0
        res=0
        for i in range(len(s)-1,-1,-1):
            if(res+s[i]>0):
                res+=s[i]
                ans+=res
            else:
                return ans 
        return ans